crates/sncast/src/helpers/scarb_utils.rs

```diff
@@ -276,6 +282,23 @@ pub fn get_package_tool_sncast(metadata: &scarb_metadata::Metadata) -> Result<&V
     Ok(tool_sncast)
 }
 
+pub fn build(config: &BuildConfig) -> Result<HashMap<String, StarknetContractArtifacts>> {
+    let mut cmd = ScarbCommand::new_with_stdio();
+    cmd.arg("build").manifest_path(&config.scarb_toml_path);
+    if config.json {
+        cmd.json();
+    }
+    cmd.run()
+        .map_err(|e| anyhow!(format!("Failed to build using scarb; {e}")))?;
+
+    let metadata = get_scarb_metadata_command(&config.scarb_toml_path)?
+        .exec()
+        .expect("Failed to obtain metadata");
+    let package = get_package_metadata(&metadata, &config.scarb_toml_path)
+        .with_context(|| anyhow!("Failed to find package"))?;
+    get_contracts_map(&metadata, &package.id)
```

Comment:
I feel like loading contracts from files is a bit of side-effect in build function.

Reply:
If you want succinctness I would one more function `build_and_load_artifacts`

Reply:
Sure, makes sense. Can I make this change in https://github.com/foundry-rs/starknet-foundry/pull/1569, as in this PR I modify build function further, and it will be much easier to do it there?

Reply:
Did just that, here's the change https://github.com/foundry-rs/starknet-foundry/pull/1569/commits/f00233d509b86ea6f9d16ab1edbdd8d67278ae4a